doc/contributing/adding-v8-fast-api.md

```diff
@@ -1,84 +1,407 @@
+  called in compliance with its function signature, so the `FastIsEven` example
+  above will only ever be called with a single argument of type `int32_t`. Any
+  calls from JavaScript whose arguments do not correspond to a fast callback
+  signature will be directed to the slow path by V8, even if the function is
+  optimized.
+* The fast callback cannot be bound directly. It must first be used to build a
+  `v8::CFunction` handle, which is passed alongside the conventional callback
+  when binding the function.
+
+## Argument and return types
+
+The following are valid argument types in Fast API callback signatures:
+
+* `bool`
+* `int32_t`
+* `uint32_t`
+* `int64_t`
+* `uint64_t`
+* `float`
+* `double`
+* `v8::Local<v8::Value>` (analogous to `any`)
+* `v8::FastOneByteString&` (analogous to `string`, but _only_ allows sequential
+  one-byte strings, which is often not useful)
+
+<!--
+Deliberately omitted:
+* `void *` (let's not go down this road...)
+* `v8::Local<v8::Object>` (this is actually treated the same as
+                           v8::Local<v8::Value> - in other words, V8 will pass
+                           any JS value in an "object" handle, whether it's an
+                           object or not, which can lead to unsafe assumptions)
+ -->
+
+The list of valid return types is similar:
+
+* `void`
+* `bool`
+* `int32_t`
+* `uint32_t`
+* `int64_t`
+* `uint64_t`
+* `float`
+* `double`
+
+<!-- * `void *` -->
+
+### Prepending a `receiver` argument
+
+V8 will always pass the "receiver" (the `this` value of the JavaScript function
+call) in the first argument position. The arguments to the JavaScript function
+call are then passed from the second position onwards.
+
+```cpp
+// Let's say that this function was bound as a method on some object,
+// such that it would be called in JavaScript as `object.hasProperty(foo)`.
+bool FastHasProperty(v8::Local<v8::Value> receiver,
+                     v8::Local<v8::Value> property,
+                     v8::FastApiCallbackOptions& options) {
+  v8::Isolate* isolate = options.isolate;
+
+  if (!receiver->IsObject()) {
+    // invalid `this` value; throw some kind of error here
   }
-  ```
 
-## Fallback to slow path
+  bool result;
+  if (!receiver.As<v8::Object>()->Has(isolate->GetCurrentContext(),
+                                      property).To(&result)) {
+    // error pending in V8, value is ignored
+    return false;
+  }
 
-Fast API supports fallback to slow path for when it is desirable to do so,
-for example, when throwing a custom error or executing JavaScript code is
-needed. The fallback mechanism can be enabled and changed from the C++
-implementation of the fast API function declaration.
+  return result;
+}
+```
 
-Passing `true` to the `fallback` option will force V8 to run the slow path
-with the same arguments.
+Even if your function binding does not need access to the receiver, you must
+still prepend it to your function arguments.
```

Comment:
This is a change from the previous guidance. It looks like V8 will no longer invoke a fast callback if the receiver argument is missing.

All examples in the codebase already follow this practice.